lib/src/matomo_dispatcher.dart

```diff
@@ -1,39 +1,81 @@
+import 'dart:convert';
+
 import 'package:flutter/foundation.dart';
 import 'package:http/http.dart' as http;
 
+import 'logger.dart';
 import 'matomo_event.dart';
 
 class MatomoDispatcher {
-  final String baseUrl;
   final String? tokenAuth;
   final http.Client httpClient;
+  final Logger? logger;
 
   final Uri baseUri;
 
-  MatomoDispatcher(this.baseUrl, this.tokenAuth, {http.Client? httpClient})
-      : baseUri = Uri.parse(baseUrl),
+  MatomoDispatcher(
+    String baseUrl,
+    this.tokenAuth, {
+    http.Client? httpClient,
+    this.logger,
+  })  : baseUri = Uri.parse(baseUrl),
         httpClient = httpClient ?? http.Client();
 
-  void send(MatomoEvent event) {
+  Future<void> send(MatomoEvent event) async {
```

Comment:
Using `async/await` instead of `.then()` in this case is purely syntactic sugar and a more modern way of handling asynchronous code. It doesn't impact performance, as the calls to `send`or `sendBatch` are still not awaited in the `_dequeue` function.

Reply:
There was a time were marking a function as async was impacted performance. Not sure if it's still the case. 

Anyway changing the type signature of a function, just to have the possibility to debug the code does not seems like a good choice.

Reply:
There is no downside to using the more modern syntax of `async/await` in this case. I would argue even that changing the function syntax to `Future<void>` is what we should do, as this accurately represents the behavior of this function: Sending a asynchronous network call to the matomo api. Changing this function signature is not a breaking change, as the `MatomoDispatcher` is only used internally in the package and not exposed to users of the package.

The change of the signature has nothing to do with easier debugging or using the logger, which was also present in the previous code.

Reply:
I agree with @JohannSchramm here, changing the signature of the method to represent its real behavior isn't that impactful as the method is only used internally. It's something that we probably should've done before.